src/neuroconv/datainterfaces/ophys/scanimage/scanimageimaginginterface.py

```diff
@@ -48,9 +53,11 @@ def __init__(
         """
         self.image_metadata = extract_extra_metadata(file_path=file_path)
 
-        if "state.acq.frameRate" in self.image_metadata:
-            sampling_frequency = float(self.image_metadata["state.acq.frameRate"])
-        else:
+        try:
+            sampling_frequency = float(
+                get_metadata_value(self.image_metadata, "state.acq.frameRate", "SI.hRoiManager.scanFrameRate")
+            )
```

Comment:
Your `util` is certainly an interesting idea

However, I think it might be overkill for this simple of a task

Could we do

```python
if "state.acq.frameRate" in self.image_metadata:
    sampling_frequency = float(self.image_metadata.get("state.acq.frameRate"])
elif "SI.hRoiManager.scanFrameRate" in self.image_metadata:
    sampling_frequency = float(self.image_metadata["SI.hRoiManager.scanFrameRate"])
else:
```

instead? This would make the separate behavior across the cases more explicit when reading the code